std/range/package.d

```diff
@@ -9594,6 +9594,324 @@ auto refRange(R)(R* range)
     foo(r);
 }
 
+/**
+Bitwise adapter over Integral type Ranges. Consumes the range elements bit by bit.
+*/
+struct Bitwise(R)
+    if (isInputRange!R && isIntegral!(ElementType!R))
+{
+    alias ER = ElementType!R;
+    alias UER = Unsigned!ER;
+
+    private R parent;
+    private UER mask;
+    private int maskPos;
+    private ER accumulator;
+
+    private enum UER bitsNum = ER.sizeof * 8;
+
+    this()(auto ref R range)
+    {
+        parent = range;
+
+        static if (hasLength!R)
+        {
+            len = parent.length * bitsNum;
+        }
+
+        initPrivates();
+
+        static if (isBidirectionalRange!R)
+        {
+            initBackPrivates();
+        }
+    }
+
+    bool empty() const
```

Comment:
const inferrence does have a workaround:

```D
bool empty(this This)() { ... }
```
This will only compile on a const wrapper range if all the calls can be made on the lower range. It's more crude than defining or not defining the method as const, but it would work.

However, my opinion is that const on ranges is useless anyway -- you can't iterate a const range. So I would recommend just not putting const here.